be/src/olap/tablet.h

```diff
@@ -27,221 +27,211 @@
-    OLAPStatus compute_all_versions_hash(const std::vector<Version>& versions,
-                                         VersionHash* version_hash) const;
-    OLAPStatus merge_tablet_meta(const TabletMeta& hdr, int to_version);
-    bool has_version(const Version& version) const;
-    void list_versions(std::vector<Version>* versions) const;
-    void mark_dropped() { _is_dropped = true; }
-    bool is_dropped() { return _is_dropped; }
-    void delete_all_files();
-    void obtain_header_rdlock() { _meta_lock.rdlock(); }
-    void obtain_header_wrlock() { _meta_lock.wrlock(); }
-    void release_header_lock() { _meta_lock.unlock(); }
-    RWMutex* get_header_lock_ptr() { return &_meta_lock; }
-    void obtain_push_lock() { _ingest_lock.lock(); }
-    void release_push_lock() { _ingest_lock.unlock(); }
-    Mutex* get_push_lock() { return &_ingest_lock; }
-    bool try_base_compaction_lock() { return _base_lock.trylock() == OLAP_SUCCESS; }
-    void obtain_base_compaction_lock() { _base_lock.lock(); }
-    void release_base_compaction_lock() { _base_lock.unlock(); }
-    bool try_cumulative_lock() { return (OLAP_SUCCESS == _cumulative_lock.trylock()); }
-    void obtain_cumulative_lock() { _cumulative_lock.lock(); }
-    void release_cumulative_lock() { _cumulative_lock.unlock(); }
-    std::string construct_dir_path() const;
-    int version_count() const;
-    const uint32_t calc_cumulative_compaction_score() const;
-    const uint32_t calc_base_compaction_score() const;
-    inline KeysType keys_type() const;
-    bool version_for_delete_predicate(const Version& version);
-    bool version_for_load_deletion(const Version& version);
-    inline const int64_t creation_time() const;
-    void set_creation_time(int64_t creation_time);
-    inline const int32_t cumulative_layer_point() const;
-    inline void set_cumulative_layer_point(const int32_t new_point);
-    AlterTabletState alter_state();
-    OLAPStatus set_alter_state(AlterTabletState state);
-    bool is_schema_changing();
-    OLAPStatus delete_alter_task();
-    OLAPStatus protected_delete_alter_task();
-    void add_alter_task(int64_t tablet_id, int64_t schema_hash,
-                        const vector<Version>& versions_to_alter,
-                        const AlterTabletType alter_type);
-    const AlterTabletTask& alter_task();
     bool is_used();
-    std::string storage_root_path_name() const;
-    std::string tablet_path() const;
-    OLAPStatus test_version(const Version& version);
-    size_t get_version_data_size(const Version& version);
-    OLAPStatus recover_tablet_until_specfic_version(const int64_t& spec_version,
-                                                    const int64_t& version_hash);
-    const std::string& rowset_path_prefix();
-    const size_t id() { return _id; }
-    void set_id(size_t id) { _id = id; }
+    inline DataDir* data_dir() const;
     OLAPStatus register_tablet_into_dir();
+    std::string dir_path() const;
```

Comment:
I think dir_path is not a good name? data_dir_path?

Reply:
OK, I remove dir_path() from tablet.h